cmd/format-xl.go

```diff
@@ -18,22 +18,58 @@ package cmd
 
 import (
 	"encoding/json"
-	"errors"
 	"fmt"
-	"reflect"
+	"io/ioutil"
 	"sync"
+
+	"encoding/hex"
+
+	"github.com/minio/minio/pkg/errors"
+	"github.com/minio/sha256-simd"
 )
 
 const (
 	// Represents XL backend.
 	formatBackendXL = "xl"
 
-	// formatXLV1.XL.Version
+	// formatXLV1.XL.Version - version '1'.
 	formatXLVersionV1 = "1"
+
+	// formatXLV2.XL.Version - version '2'.
+	formatXLVersionV2 = "2"
+
+	// Distribution algorithm used.
+	formatXLVersionV2DistributionAlgo = "CRCMOD"
```

Comment:
I won't do that, distribution algo is as much a backend format entity as other entities in this file. That is the whole reason to keep it here.

Reply:
How will you name the consts if we move to a new hash mod, lets call it "NEWHASHMOD"? you will see the problems of tying format version to distributionAlgo.

Hence it should be named in the following fashion:
const distributionAlgoCRCMOD = "CRCMOD"
const distributionAlgoNEWHASHMOD = "NEWHASHMOD"
const distributionAlgoCurrent = distributionAlgoNEWHASHMOD

The reason I say this is xlSets configuration and not backend format related is:
1) Suppose backend format changes, (ex. lock-free backend or change of the location for policy.json) it will not affect the distributionAlgo.
2) Suppose distributionAlgoCurrent changes, it does not change the format version number.

i.e distributionAlgo and format version are not related. Once a distributionAlgo is decided for a deployment it can never change even if backend format changes. An object should always hash to the same set. 

The case where distributionAlgo gets tied to backend format is if we have to migrate one distributionAlgo to another for the same backend.

Reply:
The reason i kept them in current form is because to indicate that `format.json` with V2 has the following hashing used for its sets.  It is upto you to decide to migrate or not.  Moving this out wouldn't help and may introduce more confusion on why this value exists in the first place inside format.json . I would still defer to not keep it outside into `xlSets`